### PR TITLE
Zab/server alloc reservation

### PR DESCRIPTION
**Pull request by zabbo**

### CODE REVIEW

**Comment by versity-github** [comment]
Build 214 http://jenkins-prod.vpn.versity.com:8080/job/scoutfs-run-pr/214/ git: e2edfa318e4249961bfecc460eec3e5db427e41f

**Comment by versity-github** [comment]
Build 215 http://jenkins-prod.vpn.versity.com:8080/job/scoutfs-run-pr/215/ git: e2edfa318e4249961bfecc460eec3e5db427e41f

**Comment by versity-github** [comment]
Build 217 http://jenkins-prod.vpn.versity.com:8080/job/scoutfs-run-pr/217/ git: e2edfa318e4249961bfecc460eec3e5db427e41f

**Comment by zabbo** [comment]
There were warnings during server commits that get_log_trees was exceeding its budget.   It was probably _alloc_move and we're not actively managing its consumption.   We'll try just upping the per-hold block budget, it was pretty stingy to start.

**Comment by versity-github** [comment]
Build 219 http://jenkins-prod.vpn.versity.com:8080/job/scoutfs-run-pr/219/ git: 2b1054b1535aa76fda7e79cf4b108513dbaed9b1

File: kmod/src/server.c [hunk 1]
```diff
@@ -181,61 +208,224 @@ static void stop_server(struct server_info *server)
...
+	int ret;
+
+	/* make sure holders held off once commit started */
+	spin_lock(&cusers->lock);
+	TRACE_COMMIT_USERS(sb, cusers, start);
+	if (WARN_ON_ONCE(list_empty(&cusers->applying) || cusers->nr_holders != 0))
+		ret = -EINVAL;
+	else
+		ret = 0;
+	spin_unlock(&cusers->lock);
```
Comment (bgly) [inline kmod/src/server.c:405]: can we just init ret to 0 and then not have the else?

Comment (zabbo) [reply kmod/src/server.c]: Sure, that's a bit cleaner.

**Review by bgly — APPROVED** [review]

**Comment by versity-github** [comment]
Build 220 http://jenkins-prod.vpn.versity.com:8080/job/scoutfs-run-pr/220/ git: dec741c7eca1c97283c82bebf9a57a93b2f141f3

**Comment by zabbo** [comment]
Pushed a new version that resolves conflicts with the changes to server start/stop that landed, let's see it go through a test pass.

**Comment by versity-github** [comment]
Build 221 http://jenkins-prod.vpn.versity.com:8080/job/scoutfs-run-pr/221/ git: d8231016f85f5160d8b2e0b5e1d4da081f43548a

**Pull request merged**